src/drivers/ins/ilabs/ILabs.cpp

```diff
@@ -0,0 +1,524 @@
+							   ? ORB_ID(external_ins_global_position)
+							   : ORB_ID(vehicle_global_position)) {
+	// store port name
+	strncpy(_serialDeviceName, serialDeviceName, sizeof(_serialDeviceName) - 1);
+
+	// enforce null termination
+	_serialDeviceName[sizeof(_serialDeviceName) - 1] = '\0';
+
+	if (_param_ilabs_mode.get() == ILabsMode::FULL_INS) {
+		int32_t value = 0;
+		param_set(param_find("SENS_IMU_MODE"), &value);
+		param_set(param_find("SENS_MAG_MODE"), &value);
+	}
+
+	_device_id.devid_s.devtype  = DRV_INS_DEVTYPE_ILABS;
+	_device_id.devid_s.bus_type = device::Device::DeviceBusType_SERIAL;
+
+	_px4_accel.set_device_id(_device_id.devid);
+	_px4_gyro.set_device_id(_device_id.devid);
+	_px4_mag.set_device_id(_device_id.devid);
+
+	_attitude_pub.advertise();
+	_local_position_pub.advertise();
+	_global_position_pub.advertise();
+	_sensor_baro_pub.advertise();
+	_sensor_gps_pub.advertise();
+}
+
+ILabs::~ILabs() {
+	_sensor.deinit();
+
+	perf_free(_sample_perf);
+	perf_free(_comms_errors);
+
+	perf_free(_accel_pub_interval_perf);
+	perf_free(_gyro_pub_interval_perf);
+	perf_free(_mag_pub_interval_perf);
+	perf_free(_gnss_pub_interval_perf);
+	perf_free(_baro_pub_interval_perf);
+
+	perf_free(_attitude_pub_interval_perf);
+	perf_free(_local_position_pub_interval_perf);
+	perf_free(_global_position_pub_interval_perf);
+}
+
+int ILabs::task_spawn(int argc, char *argv[]) {
+	bool error_flag = false;
+
+	int         opt_index   = 1;
+	const char *opt_arg     = nullptr;
+	int         opt_val     = 0;
+	const char *device_name = nullptr;
+
+	while ((opt_val = px4_getopt(argc, argv, "d:", &opt_index, &opt_arg)) != EOF) {
+		switch (opt_val) {
+			case 'd':
+				device_name = opt_arg;
+				break;
+
+			case '?':
+				error_flag = true;
+				break;
+
+			default:
+				PX4_WARN("Unrecognized flag");
+				error_flag = true;
+				break;
+		}
+	}
+
+	if (error_flag) {
+		return -1;
+	}
+
+	if (device_name && (access(device_name, R_OK | W_OK) == 0)) {
+		ILabs *instance = new ILabs(device_name);
+
+		if (instance == nullptr) {
+			PX4_ERR("Alloc failed");
+			return PX4_ERROR;
+		}
+
+		_object.store(instance);
+		_task_id = task_id_is_work_queue;
+
+		instance->ScheduleNow();
+
+		return PX4_OK;
+	}
+
+	if (device_name) {
+		PX4_ERR("Invalid device (-d) %s", device_name);
+
+	} else {
+		PX4_INFO("Valid device required");
+	}
+
+	return PX4_ERROR;
+}
+
+int ILabs::custom_command(int argc, char *argv[]) {
+	return print_usage("unknown command");
+}
+
+int ILabs::print_usage(const char *reason) {
+	if (reason) {
+		PX4_WARN("%s\n", reason);
+	}
+
+	PRINT_MODULE_DESCRIPTION(
+		R"DESCR_STR(
+### Description
+
+Serial bus driver for the ILabs sensors.
+
+Most boards are configured to enable/start the driver on a specified UART using the SENS_ILABS_CFG.
+After that you can use the ILABS_MODE parameter to config outputs:
+- only raw sensor output (the default)
+- additionally supplies INS data such as position and velocity estimates
```

Comment:
Mostly for the empty line, which is more prettier friendly in docs.
```suggestion
After that you can use the ILABS_MODE parameter to config outputs:

- Only raw sensor output (the default).
- Sensor output and INS data such as position and velocity estimates.
```

Reply:
Done